packages/core/src/upgrade-interface-version.ts

```diff
@@ -13,7 +14,10 @@ export async function getUpgradeInterfaceVersion(
     // The first 32 bytes represent the offset, which should be 32 for a string
     const offset = parseInt(buf.slice(0, 32).toString('hex'), 16);
     if (offset !== 32) {
-      throw new Error(`Unexpected type for UPGRADE_INTERFACE_VERSION at address ${address}. Expected a string`);
+      // Log as debug and return undefined if the interface version is not a string.
+      // Do not throw an error because this could be caused by a fallback function.
+      debug(`Unexpected type for UPGRADE_INTERFACE_VERSION at address ${address}. Expected a string`);
```

Comment:
I'd suggest adding a test to make sure this is logged if possible although it's just debug data. Surely is something you'd like to see if you're doing a production deployment.

Reply:
Thanks, this revealed that the tests were testing the wrong scenario, which is fixed in the latest two commits.

Added tests for the debug logging by using optional parameters for the relevant functions. It's not my preferred approach but couldn't get mocks to work well with this.